xio-core/src/test/java/com/xjeffrose/xio/http/EdgeProxyFunctionalTest.java

```diff
@@ -103,29 +89,48 @@ public RouteStates(Function<U, T> factory, RouteConfigs configs) {
     public EdgeProxyConfig(Config config) {
       super(config);
       List<Config> routes = (List<Config>) config.getConfigList("routes");
+
       routeConfigs =
-          new RouteConfigs<ProxyRouteConfig>((Config c) -> new ProxyRouteConfig(c), routes);
+          new RouteConfigs<>(
+              routes.stream().map(ProxyRouteConfig::new).collect(Collectors.toList()));
       allPermissions = null;
     }
   }
 
   public class EdgeProxyState extends ApplicationState {
 
-    private final RouteStates routeStates;
+    private final RouteStates<ProxyRouteState> routeStates;
     private final ProxyClientFactory clientFactory;
 
+    public <T, K, U> Collector<T, ?, Map<K, U>> toLinkedMap(
+        Function<? super T, ? extends K> keyMapper, Function<? super T, ? extends U> valueMapper) {
+
+      return Collectors.toMap(
+          keyMapper,
+          valueMapper,
+          (u, v) -> {
+            throw new IllegalStateException(String.format("Duplicate key %s", u));
+          },
+          LinkedHashMap::new);
+    }
+
     public EdgeProxyState(EdgeProxyConfig config) {
       super(config);
       clientFactory = new ProxyClientFactory(this);
       routeStates =
-          new RouteStates<ProxyRouteState, ProxyRouteConfig>(
-              (ProxyRouteConfig c) ->
-                  new ProxyRouteState(this, c, new ProxyHandler(clientFactory, c)),
-              config.routeConfigs);
+          new RouteStates<ProxyRouteState>(
+              ImmutableMap.copyOf(
+                  config
+                      .routeConfigs
+                      .stream()
+                      .map(
+                          (ProxyRouteConfig c) ->
+                              new ProxyRouteState(this, c, new ProxyHandler(clientFactory, c)))
+                      .collect(toLinkedMap(s -> s.path(), s -> s))));
```

Comment:
's' could be more descriptive as well for a variable name.

Reply:
should be `state`